### PR TITLE
Add Java 9 Optional APIs

### PR DESCRIPTION
**Pull request by treblereel**
(cherry picked from commit 9b9e888bfc5c8991425da65b0e2f12577d0ac638)

### CODE REVIEW

File: jre/javatests/com/google/j2cl/jre/java9/util/OptionalTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import com.google.j2cl.jre.java.util.EmulTestBase;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Tests for java.util.Optional Java 9 API emulation.
+ */
+public class OptionalTest extends EmulTestBase {
+  public void testIfPresentOrElse() {
```
Comment (gkdn) [inline jre/javatests/com/google/j2cl/jre/java9/util/OptionalTest.java:27]: Some of these were already tested under OptionalTest. Could you remove the tests here and if a scenario is missing could you update the other one?

File: jre/javatests/com/google/j2cl/jre/java9/util/OptionalIntTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.j2cl.jre.java.util.EmulTestBase;
+
+import java.util.OptionalInt;
+import java.util.stream.Stream;
+
+/**
+ * Tests for java.util.OptionalInt Java 9 API emulation.
+ */
+public class OptionalIntTest extends EmulTestBase {
+  public void testIfPresentOrElse() {
```
Comment (gkdn) [inline jre/javatests/com/google/j2cl/jre/java9/util/OptionalIntTest.java:27]: For consistency let's move this to java8 OptionalIntTest and follow the same patern in OptionalTest.

(And for other OptionXxxTests as well)

File: jre/javatests/com/google/j2cl/jre/java9/util/OptionalTest.java (hunk 1 above)
Comment (treblereel) [reply jre/javatests/com/google/j2cl/jre/java9/util/OptionalTest.java]: I agree, this test is redundant

File: jre/javatests/com/google/j2cl/jre/java9/util/OptionalIntTest.java (hunk 2 above)
Comment (treblereel) [reply jre/javatests/com/google/j2cl/jre/java9/util/OptionalIntTest.java]: done

**Pull request closed without merging**